moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1547,4 +1570,20 @@ void TrajectoryExecutionManager::loadControllerParams()
     }
   }
 }
+
+double TrajectoryExecutionManager::getJointAllowedStartTolerance(std::string const& jointName) const
+{
+  auto start_tolerance_it = joints_allowed_start_tolerance_.find(jointName);
+  return start_tolerance_it != joints_allowed_start_tolerance_.end() && start_tolerance_it->second > 0 ?
+             start_tolerance_it->second :
+             allowed_start_tolerance_;
+}
+
+void TrajectoryExecutionManager::updateJointsAllowedStartToleranceEmpty()
+{
+  joints_allowed_start_tolerance_empty_ =
+      std::all_of(joints_allowed_start_tolerance_.begin(), joints_allowed_start_tolerance_.end(),
+                  [](auto const& pair) { return pair.second <= 0; });
```

Comment:
I missed to commit this comment:
Just reject/remove invalid tolerance values once when setting/loading them instead of using this complicated mechanism.